classes/village/VillageUpgradeManager.php

```diff
@@ -234,29 +277,31 @@ public static function checkConstructionRequirementsMet(Village $village, string
      * @param System $system
      * @param Village $village
      * @param string $building_key
+     * @throws RuntimeException
      * @return string
      */
     public static function beginConstruction(System $system, Village $village, $building_key): string {
+        self::initialize();
         // get construction costs and time
         $materials_cost = VillageBuildingConfig::BUILDING_CONSTRUCTION_COST[$building_key][$village->buildings[$building_key]->tier + 1][WarManager::RESOURCE_MATERIALS];
         $food_cost = VillageBuildingConfig::BUILDING_CONSTRUCTION_COST[$building_key][$village->buildings[$building_key]->tier + 1][WarManager::RESOURCE_FOOD];
         $wealth_cost = VillageBuildingConfig::BUILDING_CONSTRUCTION_COST[$building_key][$village->buildings[$building_key]->tier + 1][WarManager::RESOURCE_WEALTH];
         $progress_required = VillageBuildingConfig::BUILDING_CONSTRUCTION_TIME[$building_key][$village->buildings[$building_key]->tier + 1] * 86400;
         // check if requirements met
         if (!VillageUpgradeManager::checkConstructionRequirementsMet($village, $building_key, $village->buildings[$building_key]->tier + 1)) {
-            return "Construction requirements not met!";
+            throw new RuntimeException("Construction requirements not met!");
```

Comment:
Per my other comment, optionally you can return an instance of https://github.com/elementum-games/shinobi-chronicles/blob/main/classes/ActionResult.php as a structured way to handle this that doesn't have to be try/caught. For an examples see `User#useJutsu()` 

https://github.com/elementum-games/shinobi-chronicles/blob/8bf516d26c8538b0f52eb5dee737fa987e5dadc6/classes/User.php#L1634

https://github.com/elementum-games/shinobi-chronicles/blob/8bf516d26c8538b0f52eb5dee737fa987e5dadc6/classes/SpecialMission.php#L576-L579

Up to you but personally I like the structure of it and avoiding the `try/catch` everywhere